libraries/botbuilder-ai/src/qnaMaker.ts

```diff
@@ -193,18 +193,25 @@ export class QnAMaker {
      * @param top (Optional) number of answers to return. Defaults to a value of `1`.
      * @param scoreThreshold (Optional) minimum answer score needed to be considered a match to questions. Defaults to a value of `0.001`.
      */
-    public generateAnswer(question: string|undefined, top?: number, scoreThreshold?: number): Promise<QnAMakerResult[]> {
+    public generateAnswer(context: TurnContext, top?: number, scoreThreshold?: number): Promise<QnAMakerResult[]> {
+        const question = context.activity.text;
```

Comment:
I don't think we can make this change, as it's a compatibility break in code that people are likely using. 

The answer is probably: 
1) Update the existing method as best we can. 
2) Introduce a new method with the new signature and have it work correctly.
3) Put a @depricated decerator on the old method. [https://stackoverflow.com/questions/19412660/how-should-i-mark-a-method-as-obsolete-in-js](https://stackoverflow.com/questions/19412660/how-should-i-mark-a-method-as-obsolete-in-js)

Reply:
D'oh! Yes this would be a breaking change. After talking with @Zerryth, she will instead implement the [QnAMaker.GetAnswersAsync()](https://github.com/Microsoft/botbuilder-dotnet/blob/master/libraries/Microsoft.Bot.Builder.AI.QnA/QnAMaker.cs#L108) from the dotnet SDK  in JS and mark QnAMaker.generateAnswer() as deprecated.

Reply:
Ok. 

@stevengum found that in the C# there's a method `GetAnswersAsync()` that does not exist in the TS version

I'll make a new method `getAnswers()` in TS, deprecate `generateAnswers()`, and write tests for `getAnswers()` method

It `generateAnswer()` will be left as it originally, since it can't `emitTraceInfo()` without accepting the breaking change `TurnContext` parameter type